velox/vector/tests/VectorTest.cpp

```diff
@@ -3101,5 +3101,240 @@ TEST_F(VectorTest, appendNulls) {
   // Append negative.
   EXPECT_ANY_THROW(rowVector->appendNulls(-1));
 }
+
+TEST_F(VectorTest, debug) {
+  auto base = makeFlatVector<int32_t>({1, 2, 3});
+  auto other = makeNullableFlatVector<int32_t>({1, std::nullopt, 1});
+  ASSERT_TRUE(
+      base->equalValueAt(
+              other.get(), 0, 0, CompareFlags::NullHandlingMode::StopAtNull)
+          .value());
+  ASSERT_FALSE(
+      base->equalValueAt(
+              other.get(), 0, 1, CompareFlags::NullHandlingMode::NoStop)
+          .value());
+  ASSERT_TRUE(
+      base->equalValueAt(
+              other.get(), 0, 2, CompareFlags::NullHandlingMode::StopAtNull)
+          .value());
+}
+
+TEST_F(VectorTest, nullsCompare) {
```

Comment:
This test is very long and it is hard to tell what paths it is testing. Would you explain the design of this test? Would it be possible to break it up maybe into testing primitive types and complex types or flat and encoded vectors?

Reply:
Yes, it is too long and hard to review, I will break this into multiple cases according to your guidance.